### DISCUSSION TITLE
EdgeX cannot communicate with Device Service

### DISCUSSION THREAD

**Discussion by sounak-1989** [edgexfoundry/community#113] [General]
Hi guys,

I am using EdgeX Kamakura. I am running EdgeX from EdgeX compose using Docker. I am running the device service natively from source. So it is a hybrid development environment. 

The device service is able to register devices, profiles and provision watchers to core-metadata. But when I am sending a command or querying a resource using core-data or core-command, I am unable to access it. I am getting the following error.

`level=ERROR ts=2023-05-23T22:44:02.038901461Z app=core-command source=http.go:47 X-Correlation-ID=3e2d68a2-35d2-4704-9cf3-dcd3c5f40d20 msg="failed to send a http request -> Get \"http://0.0.0.0:59991/api/v2/device/name/IOCard2031666/VInP1\": dial tcp 0.0.0.0:59991: connect: connection refused"`

Although when I query the same endpoint using Postman I am able to get a response. There has been some kind of communication issue between the microservices running inside docker and the device service running natively. Any idea what can cause this issue?

Thanks,
Sounak

**Comment by bnevis-i** [comment]
For some reason the service is registering its listening host as `0.0.0.0` where it should be `localhost` for a snap, or `edgex-device-whatever` (whatever you have configured for hostname and SERVICE_HOST) for docker.  `0.0.0.0` means listen on all interfaces, but `0.0.0.0` as a target is indecipherable because the outgoing network interface is ambiguous.

**Comment by lenny-goodell** [comment]
@sounak-1989 , you must set the Service.Host in your configuration to the IP address of your system. This is the only way the services running in the Docker network can find your service running outside docker network.

You will need to do a clean restart of EdgeX stack so that the DB volume has been removed. 
`docker-compose -f <compose-file> down -v`
Otherwise Core Metadata will still have your previous Host information in the DeviceService object.

**Reply by sounak-1989** [reply]
Ahh, got it. Thanks for the tip. I was wondering how to do a clean restart of EdgeX. Before Kamakura I used to prune the docker volumes and it used to work.
I'll give it a try and lett you know.

**Reply by sounak-1989** [reply]
@lenny-intel This works. Thanks for the solution.
